### ISSUE TITLE
`Feature` fields getters and setters should accept field indexes

### ISSUE THREAD

**Issue by lnicola** [georust/gdal#500]
> I was recently looking at a small program that spent 30% of its CPU time in strcasecmp.

To be honest, I wish they'd only take the field index and never the name, but that's might really upset some users.

**Comment by ChristianBeilschmidt** [comment]
Well, a resolution from name to index is a convenience thing, I guess. There could be two separate methods or one that accepts both. But you definitely need the opportunity to use the names.

So, what is your suggestion here?

**Comment by lnicola** [comment]
So we can add `fn set_field_by_index(&self, idx: usize, value: &FieldValue)`, `fn set_field_integer_by_index(&self, idx: usize, value: i32)` and about 9 more.

But I'm worried that users will just do:

```rust
for feature in layer.features() {
    feature.set_field_integer("foo", 20);
}
```

instead of:

```rust
let foo_col = layer.field_index("foo")?;
for feature in layer.features() {
    feature.set_field_integer_by_index(foo_col, 20);
}
```

just because it's easier to find.

**Comment by ChristianBeilschmidt** [comment]
Yeah, either it would be `…by_index` and `…by_name` separately or you just have to keep strongly typing it to size :smile: .
